### PR TITLE
feat: Parameter transformation

### PR DESCRIPTION
**Pull request by Natoandro**
Enable parameter transformation with the `.apply()` method.
It has more or less the same logic as `.reduce()` with the ability to flatten the input type.

#### Motivation and context

This feature enables simpler APIs (input types) on top of runtimes (e.g.: prisma).

#### Migration notes

_No changes needed_.

### Checklist

- [x] The change come with new or modified tests
- [x] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: typegate/src/engine/planner/parameter_transformer.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  ): string {
+    const varName = this.#createVarName();
+    this.#collector.push(`const ${varName} = {};`);
+
+    for (const [key, node] of Object.entries(fields)) {
+      const path = this.#path;
+      path.push(key);
+      const propVarName = this.#compileNode(node);
+      path.pop();
+      this.#collector.push(`${varName}["${key}"] = ${propVarName};`);
```
Comment (afmika) [inline typegate/src/engine/planner/parameter_transformer.ts:104]: Can't we do these code gynmastics without strings? This approach seems susceptible for injection as it is directly tied to the arg planner.

File: typegate/tests/params/apply_test.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      })
+      .on(e);
+  });
+
+  await t.should("work with renamed params", async () => {
+    await gql`
+      query {
+        flattened(a1: 1, a2: 2, b11: 3, b12: 4, b2: 5) {
+          a { a1 a2 }
+          b { b1 { b11 b12 } b2 }
```
Comment (afmika) [inline typegate/tests/params/apply_test.ts:32]: Neat 👍

File: typegate/src/engine/planner/parameter_transformer.ts (hunk 1 above)
Comment (Natoandro) [reply typegate/src/engine/planner/parameter_transformer.ts:104]: Thanks :+1:

File: typegate/tests/params/apply.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                {
+                    "a": t.list(t.integer()),
+                }
+            )
+        ).apply(
+            {
+                "a": [
+                    g.as_arg("first"),
+                    g.as_arg("second"),
+                ]
```
Comment (afmika) [inline typegate/tests/params/apply.py:95]: Looking great so far!
Maybe add tests for `g.as_arg("..")` at the end the following edgecase edgecases at the end:
1. A nested arrays at different level `[[g.as_arg("A")], g.as_arg("B")]` 
2. Nested arrays from some items of an array of objects `[ [ { a: {b: [ g.as_arg("B") ] } ], ... ]`
They both look tricky imo.

**Review by Yohe-Am — DISMISSED** [review]

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**